### ISSUE TITLE
Ability to enable only [all] recommended extensions

### ISSUE THREAD

**Issue by n3rd4i** [microsoft/vscode#61723]
The ability to only **enable** the **recommended extensions** for the _specific workspace_ [folder].

**Comment by sandy081** [comment]
@n3rd4i Do you mean to install or enable them only for the workspace that has recommendations?

**Comment by n3rd4i** [comment]
> 
> 
> Do you mean to install or enable them only for the workspace that has recommendations?

Sort answer, maybe both.

1. Enable them (if they are disabled) for the workspace where they are recommended. 
1.1 But to be able to do so with one command (enable recommended extensions) like:
   -- e.g. Existing command: `Enable Auto Updating Extensions` (which maps to internal command)
   -- New command in that listing: `Enable Recommended Extensions` (which maps to internal command)
1. Additionally, It would be helpful if they where also installed if they are not there already.
1. Also, always when extensions are installed, they are enabled globally, never they are enabled for the workspace where they are installed.

**Comment by n3rd4i** [comment]
> [...] when recommendations are installed, enable only for this workspace automatically?

That sounds very helpful, I would suggest that, on top of the other points I made earlier.

Additionally, maybe also allowing to **disable recommended extensions** could prove useful to still keep the global enabled ones untouched.

**Comment by whut** [comment]
@sandy081 

> So you mean when recommendations are installed, enable only for this workspace automatically?

Yes! With that single thing it would be easy to maintain different sets of extensions per workspace and keep them in source control.

Here is the workflow:

1. First disable all extensions except selected few that are used always. This is a required step to use this workflow - this should be documented on [Recommended Extensions](https://code.visualstudio.com/docs/editor/extension-gallery#_recommended-extensions) section in Extensions user guide to make users familiar with this approach.

2. When opening some workspace with recommended extensions that I have already installed, Code **automatically enables recommended extensions for current workspace**. I believe this should win even over manually "disabled per workspace" extension, as the same behavior is with settings: workspace settings win over user settings. Note that there is no disabling extensions automatically (that kind of idea is proposed in #40239, IMO it would be irritating) nor automatic install, user is still in control here.

3. When opening some workspace with recommended extensions and I do not have them already installed, Code prompts to install them, and with #51658 I am able to right on the spot enable them only in current workspace. Even without #51658 it is OK, I just install them, and disable, then reload window to trigger behavior from above point.

In the end we have system that is simple, automatic, **source controlled so sharable with the team** and still **keeps user in control on which extension are installed**, and allows to have additional extensions enabled

**Comment by ArulDhandapani** [comment]
Reading all the thread related with this topic, i assume everyone looking similar to below feature what I described with example.

This has to go to ".code-workspace" config file to enable other users as zero config/setup effort in a project, they just open and start coding.

Example File: project.code-workspace
	{
		"folders": [
			{
				"path": "."
			}
		],
		"settings.user": "false", <-Enable/Disable user settings located under 'AppData\Roaming\Code\User\settings.json' in a system
		"settings.folder": "false", <-Enable/Disable folder settings located under '.vscode\settings.json' in a project
		"extensions": { <-Extension config
			"installed": "false", <-Enable/Disable user installed extension in their system
			"[load]": { <-Re-load VSCode when "open workspace"
				"required": [ <-Listed below extension auto-install if user doesn't have in their system without prompt
					"ms-python.python",
					"sonarsource.sonarlint-vscode",
					"eamodio.gitlens"
				],
				"optional": [ <-Listed below extension prompt to user to install, if user doesn't have in their system
					"euskadi31.json-pretty-printer",
					"silverbulleters.sonarqube-inject"
				],
			},
		}
	}

**Comment by neodon** [comment]
I appreciate the discussion here and all the hard work done by the contributers to this project.  Please don't take my criticism/complaint here too harshly.  Overall I love VSCode--it is amazing.  This is a quality-of-life improvement request.

I like the existing feature for specifying and installing recommended extensions.  It has helped me sync expectations with other devs on the extensions to use.  It feels incomplete, however.

## Enabling recommended extensions per workspace is tedious

This GitHub issue calls out the missing piece in the recommended extensions feature(s).  Currently, one has to manually enable recommended extensions one at a time when opening any code workspace for the first time on every machine.  For me and my colleagues it's usually about 10-15 extensions each time (that's not too crazy, is it?).  

__*I'd like some way to just enable all workspace recommended extensions only in the current workspace, and not globally.*__

## Current tedious process

This applies to opening a code workspace for the first time in any directory on any machine.

1. Open the code workspace.
2. Click the button to install recommended extensions.  Probably reload.
3. Now go and disable all those new extensions so they won't start loading globally for other workspaces.  Possibly reload again.  __Side note: I'd rather VSCode not assume I want every extension to be enabled globally by default.  It's surprising behavior in this case and accumulating extensions only needed by certain workspaces seems problematic.__
4. Go to the workspace recommended extension list and manually enable them for the current workspace only, one-by-one.  Maybe one more reload.

## Suggestions to consider

1. I would be ecstatic if there was a setting to automatically enable recommended extensions just for the current workspace (not globally though, and only after the user initially gives permission for that, probably).

2. I'd even be happy with just a command like "Extensions: Enable Recommended Extensions For This Workspace".

3. A third option would be to add a way to perform an action on all the extensions currently listed (or selected even) in the extensions view.  So if I filter by `@recommended`, I could then perform the action "Install/Enable these extensions for the current workspace".

---

Thank you for taking the time to read through this response.

**Comment by jd-0001** [comment]
I am also waiting for this feature.

Here's my workflow: https://github.com/microsoft/vscode/issues/128030

Additionally, If I create a new project for let's say reproducing some issue I want syntax highlighting for that workspace only and want to enable extensions for that workspace only rather than enabling globally. Enabling them one by one by click on the Gear icon is not I want.

**Comment by jd-0001** [comment]
Any updates?

**Comment by Junkher** [comment]
4 year...

**Comment by Junkher** [comment]
Any updates?

**Comment by ImaniAN** [comment]
### The way I need this feature right now ... 

Sometimes when you working in different workspaces that have code that uses different languages, Frameworks, Platforms or Libraries(the VS Code workspaces)... 

You have to enable and disable extensions according to that specific workspace. 

Would be pretty cool if when you open a workspace, you are asked if you want to:
 
_**firstly**: install all recommended workspace extensions
**Secondly**: Disable all extensions not in the recommended workspace extensions array/list
**Lastly**: Enable the recommended workspace extensions_

### SIDE NOTE:
users should be allowed to create a list of **"globally available extensions"**:
**_def_**: A list of installed extensions that can be enabled one after another

Use Case (Personal): 
Regardless of workspace, repo, language, Framework, Platform or Library or code base if I am using VS Code, I have _Kite AutoComplete AI Code_ and _TODO Highlight_ enabled, because in my opinion I will always need them regardless of what I am working on. 

Even if I disable all extensions, it would be helpful to enable extensions that, regardless of the recommended workspace extensions can be enabled as a collective  _(i.e/ AKA => "globally available extensions")_

**Comment by avaranze** [comment]
Hello?

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by jd-solanki** [comment]
This is useful if someone is working with multiple languages. I don't want to use python extension in TS code and vice versa.

To prevent unwanted use of resources we disable the extensions and we don't have an easy way to enable them back on language-specific projects.

**Comment by sandy081** [comment]
For language specific projects, please check out this feature - https://github.com/microsoft/vscode/issues/116740
